readthedocs/oauth/tasks.py

```diff
@@ -226,3 +237,518 @@ def attach_webhook(project_pk, user_pk=None, integration=None, **kwargs):
         },
     )
     return False
+
+
+class GitHubAppWebhookHandler:
+    """
+    Handle GitHub App webhooks.
+
+    All event handlers try to create the installation object if it doesn't exist in our database,
+    except for events related to the installation being deleted or suspended.
+    This guarantees that our application can easily recover if we missed an event
+    in case our application or GitHub were down.
+    """
+
+    def __init__(self, data: dict, event: str):
+        self.data = data
+        self.event = event
+
+    @cached_property
+    def gh_app_client(self):
+        return get_gh_app_client()
+
+    def handle(self):
+        # Most of the events have an installation object and action.
+        installation_id = self.data.get("installation", {}).get("id", "unknown")
+        action = self.data.get("action", "unknown")
+        event_handlers = {
+            "installation": self._handle_installation_event,
+            "installation_repositories": self._handle_installation_repositories_event,
+            "installation_target": self._handle_installation_target_event,
+            "push": self._handle_push_event,
+            "pull_request": self._handle_pull_request_event,
+            "repository": self._handle_repository_event,
+            "organization": self._handle_organization_event,
+            "member": self._handle_member_event,
+            "github_app_authorization": self._handle_github_app_authorization_event,
+        }
+        log.bind(
+            installation_id=installation_id,
+            action=action,
+            event=self.event,
+        )
+        if self.event in event_handlers:
+            log.debug("Handling event")
+            event_handlers[self.event]()
```

Comment:
This needs a `return` here, otherwise, the following lines with the `raise` will always be executed.

```suggestion
            return event_handlers[self.event]()
```

Reply:
Actually, probably is easier to read with the negation:

```python
if self.event not in event_handlers:
  raise ValueError

return event_handlers[self.event]()
```